lib/mocha/parameters_matcher.rb

```diff
@@ -28,7 +29,11 @@ def mocha_inspect
     end
 
     def matchers
-      @expected_parameters.map(&:to_matcher)
+      if (last_parameter = @expected_parameters.last).is_a?(Hash)
+        @expected_parameters[0...-1].map(&:to_matcher) + [ParameterMatchers::LastPositionalHash.new(last_parameter)]
```

Comment:
Alternatively: override `Hash#to_matcher` and pass in a `last: true` arg? What do you think @floehopper? I think it might match the encapsulation better, though it makes the behaviour less obvious.

Reply:
I think I might prefer your alternative solution - or at least I like the sound of it enough to want to see how it looks! 😄

Reply:
Attempted here https://github.com/freerange/mocha/pull/534/commits/cf19d366fc57a94ddd8e4d79ab585b5cddd529fd, but not as elegant as I thought it'd be >_<. 

I also tried something like:
```ruby
@expected_parameters.last.mark_last_argument # sets an instance variable
@expected_parameters.map(&:to_matcher) # checks instance variable to know whether it's the last arg
```
But the above fails if the parameter is frozen. I think we could do something like wrapping in a Parameter object but it felt like it would be more complex for little benefit.